### PR TITLE
[IOTDB-5725] Make internal report recording measurements asynchronous

### PR DESCRIPTION
**Pull request by OneSizeFitsQuorum**
The flush time has been greatly reduced, and the system reject error has disappeared.
image

### CODE REVIEW

**Review by SpriCoder — APPROVED** [review]
LGTM~

File: server/src/main/java/org/apache/iotdb/db/service/metrics/IoTDBInternalLocalReporter.java [hunk 1]
```diff
@@ -150,39 +150,43 @@ public ReporterType getReporterType() {
...
-      List<Object> values = new ArrayList<>();
-      for (Map.Entry<String, Object> entry : valueMap.entrySet()) {
-        String measurement = entry.getKey();
-        Object value = entry.getValue();
-        measurements.add(measurement);
-        types.add(inferType(value));
-        values.add(value);
-      }
-      ByteBuffer buffer = SessionUtils.getValueBuffer(types, values);
+    service.execute(
```
Comment (xingtanzjr) [inline server/src/main/java/org/apache/iotdb/db/service/metrics/IoTDBInternalLocalReporter.java:153]: From the call stack, it seems that the `service.execute()` will be called inside the thread pool, which has potential risk leading dead lock for thread pool `service`

Comment (xingtanzjr) [reply server/src/main/java/org/apache/iotdb/db/service/metrics/IoTDBInternalLocalReporter.java:153]: Yes, you're correct. Actually my point of previous comment is that there is `potential risk` in current implementation at the same time it won't lead to dead-lock. This is just a talk about whether submit a new task inside a thread pool to itself is a the best practice or not.

Comment (OneSizeFitsQuorum) [reply server/src/main/java/org/apache/iotdb/db/service/metrics/IoTDBInternalLocalReporter.java:153]: > This is just a talk about whether submit a new task inside a thread pool to itself is a the best practice or not.

Comment (OneSizeFitsQuorum) [reply server/src/main/java/org/apache/iotdb/db/service/metrics/IoTDBInternalLocalReporter.java:153]: > This is just a talk about whether submit a new task inside a thread pool to itself is a the best practice or not.
Yeah, I don't want to do that either. But if the cost of best practices is to introduce an extra thread, I'm inclined to start from a thread count reduction perspective for now (we want to reduce thread count in IoTDB within the next year).

> shall we add some limitation to it such as change the method name to something like xxxxxAsync.
Good advice. It's been changed.

**Review by xingtanzjr — APPROVED** [review]

**Pull request merged**